### PR TITLE
FSET-1542: Provide flag for turning off test assessor generation

### PR DESCRIPTION
**Pull request by fayimora**
This is an initial stab at making the user generation dynamic. At the moment, I've used a flag for this `?generateDefaultAssessor=false`. I won't be surprised if we will want to add the same flag for the other types of roles. 

I can either make the others dynamic too via query params OR we provide a flag to turn them all on/off OR we just do for the one flag(assessors) and add for others as the need arises.

Thoughts?

_p.s I'm making this change because it is required for a test in FSET-1542_

### CODE REVIEW

**Comment by fayimora** [comment]
Decided to go with turning them all on/off via one flag. Commit enroute

File: app/controllers/testdata/TestDataGeneratorController.scala [hunk 1]
```diff
@@ -46,8 +46,8 @@ trait TestDataGeneratorController extends BaseController {
     Ok("OK")
   }
 
-  def clearDatabase() = Action.async { implicit request =>
-    TestDataGeneratorService.clearDatabase().map { _ =>
+  def clearDatabase(generateDefaultRoles: Boolean) = Action.async { implicit request =>
```
Comment (mgilgar) [inline app/controllers/testdata/TestDataGeneratorController.scala]: I think instead of generateDefaultRoles is more like generateDefaultUsers.

Comment (fayimora) [inline app/controllers/testdata/TestDataGeneratorController.scala]: 👍

File: app/services/testdata/TestDataGeneratorService.scala [hunk 2]
```diff
@@ -37,10 +37,18 @@ object TestDataGeneratorService extends TestDataGeneratorService {
 
 trait TestDataGeneratorService extends MongoDbConnection {
 
-  def clearDatabase()(implicit hc: HeaderCarrier): Future[Unit] = {
+  def clearDatabase(generateDefaultUsers: Boolean)(implicit hc: HeaderCarrier): Future[Unit] = {
     for {
       _ <- db().drop()
       _ <- AuthProviderClient.removeAllUsers()
+      _ <- generateUsers() if generateDefaultUsers
+    } yield {
```
Comment (benjstephenson) [inline app/services/testdata/TestDataGeneratorService.scala]: You don't need these braces here

**Comment by fayimora** [comment]
@henricook I think this is ready 😉

**Comment by henricook** [comment]
Love it.

**Pull request merged**